lib/dotdiff/element_handler.rb

```diff
@@ -9,27 +9,28 @@ def initialize(driver, elements = DotDiff.xpath_elements_to_hide)
 
     def hide
       elements.each do |xpath|
-        if element_exists?(xpath)
-          driver.execute_script("#{js_element(xpath)}.style.visibility = 'hidden'")
-        end
+        driver.execute_script(script(xpath, :hidden))
       end
     end
 
     def show
       elements.each do |xpath|
-        if element_exists?(xpath)
-          driver.execute_script("#{js_element(xpath)}.style.visibility = ''")
-        end
+        driver.execute_script(script(xpath, :''))
       end
     end
 
-    def js_element(xpath)
-      "document.evaluate(\"#{xpath}\", "\
-      "document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue"
-    end
+    def script(xpath, visibility)
+      xpath += if visibility == :hidden
+        "[not(contains(@style, 'visibility'))]"
+      else
+        "[contains(@style, 'visibility: hidden')]"
```

Comment:
This is dependent on a specific format of the style as someone could have `style="visibility:hidden"` which this wouldn't match. Not at a big deal at this point but something to be mindful of